OpenRA.Mods.Common/Activities/Move/VisualMoveIntoTarget.cs

```diff
@@ -35,45 +35,41 @@ protected override void OnFirstRun(Actor self)
 			targetStartPos = target.Positions.PositionClosestTo(self.CenterPosition);
 		}
 
-		public override Activity Tick(Actor self)
+		public override bool Tick(Actor self)
 		{
 			if (IsCanceling || target.Type == TargetType.Invalid)
-				return NextActivity;
+				return true;
 
 			if (mobile.IsTraitDisabled || mobile.IsTraitPaused)
-				return this;
+				return false;
 
 			var currentPos = self.CenterPosition;
 			var targetPos = target.Positions.PositionClosestTo(currentPos);
 
 			// Give up if the target has moved too far
 			if (targetMovementThreshold > WDist.Zero && (targetPos - targetStartPos).LengthSquared > targetMovementThreshold.LengthSquared)
-				return NextActivity;
+				return true;
 
 			// Turn if required
 			var delta = targetPos - currentPos;
 			var facing = delta.HorizontalLengthSquared != 0 ? delta.Yaw.Facing : mobile.Facing;
 			if (facing != mobile.Facing)
 			{
-				var turn = ActivityUtils.RunActivity(self, new Turn(self, facing));
-				if (turn != null)
-					QueueChild(turn);
-
-				return this;
+				mobile.Facing = Util.TickFacing(mobile.Facing, facing, mobile.TurnSpeed);
```

Comment:
Why not queue the ChildActivity?

Again, would be ideal if the logical changes can be kept in a separate commit from the mechanical ones.

Reply:
It's to prevent the turn from overshooting.